### PR TITLE
Score based common source yaml 2026

### PR DESCRIPTION
**Pull request by wowtor**

### CODE REVIEW

File: examples/score-based-common-source.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+# 2. PATH TO DATA AND HOW TO PARSE IT
+# An example dataset of LA-ICP-MS measurements of elemental concentration from floatglass is provided as a default dataset. For more
+# information on the dataset, see: https://github.com/NetherlandsForensicInstitute/elemental_composition_glass
+#
+# This example is only provided to give working default values for the information you need to enter for your own data. 
+# You should give a path to your data. If your data is in the default format, the default parser should work.
+# If your data is not in the default format you can either write a new parser (the hard option) or, if possible, reformat your data (the easy option).
+# If you want to know how to use the default csv parser, go here. <TODO>
+
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: in the comments link naar de default csv geven, zodat ze het zelf kunnen downloaden.
Zodat de data-provider in de vorm kan komen die het meest logisch is als je zelf je eigen csv meebrengt

File: examples/score-based-common-source.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+data_provider:
+  method: feature_data_csv_http
+  url: https://raw.githubusercontent.com/NetherlandsForensicInstitute/elemental_composition_glass/main/training.csv
+  instance_id_column: id
+  source_id_column: Item
+  ignore_columns:
+    - Piece
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml:52]: uitleg over die columns

File: examples/score-based-common-source.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+# An example dataset of LA-ICP-MS measurements of elemental concentration from floatglass is provided as a default dataset. For more
+# information on the dataset, see: https://github.com/NetherlandsForensicInstitute/elemental_composition_glass
+#
+# This example is only provided to give working default values for the information you need to enter for your own data. 
+# You should give a path to your data. If your data is in the default format, the default parser should work.
+# If your data is not in the default format you can either write a new parser (the hard option) or, if possible, reformat your data (the easy option).
+# If you want to know how to use the default csv parser, go here. <TODO>
+
+data_provider:
+  method: feature_data_csv_http
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: ik zou parsers 'parser' noemen

File: examples/score-based-common-source.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  url: https://raw.githubusercontent.com/NetherlandsForensicInstitute/elemental_composition_glass/main/training.csv
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: suggestion:
"""
  file: default_example.csv
"""

File: examples/score-based-common-source.yaml (hunk 3 above)
Comment (deflaaf) [reply examples/score-based-common-source.yaml]: suggestion:
  method: feature_data_csv_parser
(en hetzelfde voor andere parsers)

File: examples/score-based-common-source.yaml (hunk 1 above)
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: The parser expects a csv with columns with a header row. To help the parser, you need to specify the meaning of the columns. 
You'll need a source_id-column, where labels for each of the sources of your data are given.
You need an instance_id-column, where labels for each of the instances exist. This may be a measurement ID, a filename (for voice recordings or face images, for instance), or something else that is meaningful to you and is distinct for all instances in your set, across different sources.
Next, you can specify the columns to ignore. These may contain metadata that is helpful for you, but do not need to be interpreted as features or measurements. 
All other columns that are found in your file will be interpreted as feature data. Note the data needs to be numerical. If not, the process will crash.

File: examples/score-based-common-source.yaml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+data_splits:             # <-- adapt this name to your own preference, you will need it later when setting up the validation experiment
+  strategy: multiclass_cross_validation
+  folds: 5
+
+# Alternatively, you may choose a simple split, where some data is assigned the role of training data and some is assigned the role of validation data.
+# If you want that, you use 'multiclass_train_test_split' for 'strategy' and you specify what the size of the test data with 'test_size'.
+# All the rest will be used as training data. The test size can be a float between 0 and 1, which will be used as a percentage.
+# It can also be an integer, that will be used as an absolute number of sources.
+# You also specify a seed for the random assignment of sources to the train or test category.
+# If you want this, uncomment the next 3 lines and comment out the two lines above.
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: suggestion:

If you want this, uncomment the next four lines and comment out the three lines above.

File: examples/score-based-common-source.yaml [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+#data_splits:
+#  strategy: multiclass_train_test_split
+#  seed: 42
+#  test_size: 0.3
+
+
+# 4. LR SYSTEM DESIGN
+# Next, the score-based common source LR system will be defined.
+
+score_based_lr_system:        # <-- this is a custom name of your LR system that you can adapt. It will be referred to later.
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: een idee om sommige van de termen te markeren die door jezelf kunnen worden bedacht: "my_" toevoegen. Dus dit wordt 'my_score_based_lr_system', en elders ook.

File: examples/score-based-common-source.yaml [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+# To find out how more complex designs should be set up, see here LINK <TODO>
+# For now, we define a single simple validation experiment:
+
+# 5. THE VALIDATION EXPERIMENT
+
+experiments:                             
+  - name: my_validation_run              # This is the name of the experiment, use your own preferred value.
+    strategy: single_run                 # For a single validation experiment, use 'single_run'.
+    lr_system: ${score_based_lr_system}  # This refers to the lr system defined above, make sure to use the name you used above.
+    data:
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml:144]: suggestion:
"""
    data_provider: ${my_data_provider}
    data_splits: ${my_data_splits}
"""

File: examples/score-based-common-source.yaml [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+# It is possible to set up experiments where multiple design choices are involved, to enable parameter and model selection.
+# To find out how more complex designs should be set up, see here LINK <TODO>
+# For now, we define a single simple validation experiment:
+
+# 5. THE VALIDATION EXPERIMENT
+
+experiments:                             
+  - name: my_validation_run              # This is the name of the experiment, use your own preferred value.
+    strategy: single_run                 # For a single validation experiment, use 'single_run'.
+    lr_system: ${score_based_lr_system}  # This refers to the lr system defined above, make sure to use the name you used above.
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: suggestion:

lr_system: ${my_score_based_lr_system}

File: examples/score-based-common-source.yaml (hunk 6 above)
Comment (deflaaf) [reply examples/score-based-common-source.yaml]: suggestion:
my_score_based_lr_system:        # <-- this is a custom name of your LR system that you can adapt. It will be referred to later.

File: examples/score-based-common-source.yaml [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    - Piece
+
+
+#3. ASSIGNING TEST AND TRAIN DATA ROLES
+# This will describe which data is used to train the system and which to test (validate) the system.
+
+# You may choose a cross-design, in which the data is divided into folds. Each fold is used as validation data sequentially,
+# and for each of the folds an LR system is trained with all the other folds. The outcomes for all folds are then pooled into one set.
+
+data_splits:             # <-- adapt this name to your own preference, you will need it later when setting up the validation experiment
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: suggestion:
my_data_splits:             # <-- adapt this name to your own preference, you will need it later when setting up the validation experiment

File: examples/score-based-common-source.yaml [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+# 2. PATH TO DATA AND HOW TO PARSE IT
+# An example dataset of LA-ICP-MS measurements of elemental concentration from floatglass is provided as a default dataset. For more
+# information on the dataset, see: https://github.com/NetherlandsForensicInstitute/elemental_composition_glass
+#
+# This example is only provided to give working default values for the information you need to enter for your own data. 
+# You should give a path to your data. If your data is in the default format, the default parser should work.
+# If your data is not in the default format you can either write a new parser (the hard option) or, if possible, reformat your data (the easy option).
+# If you want to know how to use the default csv parser, go here. <TODO>
+
+data_provider:
```
Comment (deflaaf) [inline examples/score-based-common-source.yaml]: suggestion:

my_data_provider:                        # <-- adapt this name to your own preference, you will need it later when setting up the validation experiment

**Review by deflaaf — APPROVED** [review]
Hey, er zitten nog wat TODO's in the text, maar het is prima om daar nog even mee te wachten. Vaak moet daar iets aparts voor geschreven worden, dat is iets meer involved dan we nu even snel kunnen doen.

File: examples/score-based-common-source.yaml [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+# Below is a complete working example. If you want to adapt the setup to your own needs, you may want to specify:
+#
+# 1. Output folder
+# 2. Path to data & how to parse it
+# 3. How to assign training and testing roles to your data
+# 4. The design of the lr system itself
+# 5. The validation experiment you run for your lr system
+#
+# When you are done, you can run it all using:
+# pdm run lir --setup examples/score-based-common-source.yaml
```
Comment (wowtor) [inline examples/score-based-common-source.yaml]: Dit helemaal bovenaan
